apps/web/app/features/stacking/hooks/use-pool-info.tsx

```diff
@@ -45,49 +45,54 @@ export function usePoolInfo(poolSlug: PoolSlug | null) {
     !getStatusQuery.data ||
     getPoolAddressQuery.isError ||
     !getPoolAddressQuery.data ||
-    stackingTrackerPool.isError ||
-    !stackingTrackerPool.data ||
     !stxMarketData;
 
   const pool = useMemo(() => poolSlug && getPoolFromSlug(poolSlug), [poolSlug]);
 
   const poolRewardProtocolInfo = useMemo(() => {
-    if (isLoading || isError || !pool) {
+    if (isLoading || isError || !pool || !poolSlug) {
       return null;
     }
 
     const poxAddress = getStatusQuery.data?.stacked
       ? getStatusQuery.data.details.pox_address
       : null;
 
-    const tvl = stackingTrackerPool.data.lastCycle?.pool?.stacked_amount
-      ? toHumanReadableStx(stackingTrackerPool.data.lastCycle?.pool?.stacked_amount, 0)
-      : '100,000,000 STX';
-
-    const tvlBaseCurrencyAmount = baseCurrencyAmountInQuote(
-      createMoneyFromDecimal(stackingTrackerPool.data.lastCycle?.pool?.stacked_amount ?? 0, 'STX'),
-      stxMarketData
-    );
-    const tvlUsd = i18nFormatCurrency(tvlBaseCurrencyAmount);
-
-    const title = stackingTrackerPool.data.entity.name || pool.name;
-    const url = stackingTrackerPool.data.entity.website || pool.url;
-    const apr = stackingTrackerPool.data.entity.apr
+    const tvl =
+      stackingTrackerPool.data?.lastCycle?.pool?.stacked_amount !== undefined
+        ? toHumanReadableStx(stackingTrackerPool.data.lastCycle?.pool?.stacked_amount, 0)
+        : null;
+
+    const tvlBaseCurrencyAmount =
+      stackingTrackerPool.data &&
+      baseCurrencyAmountInQuote(
+        createMoneyFromDecimal(
+          stackingTrackerPool.data.lastCycle?.pool?.stacked_amount ?? 0,
+          'STX'
+        ),
+        stxMarketData
+      );
+    const tvlUsd = tvlBaseCurrencyAmount && i18nFormatCurrency(tvlBaseCurrencyAmount);
+
+    const title = stackingTrackerPool.data?.entity.name || pool.name;
+    const url = stackingTrackerPool.data?.entity.website || pool.url;
```

Comment:
This is fine for now, but when we refactor, I'd love you to think about how we can perform all of this logic in a way that is not tied so closely to the views